src/atlas/interpolation/method/sphericalvector/SphericalVector.cc

```diff
@@ -0,0 +1,296 @@
+
+  // Get matrix data.
+  const auto nRows = matrix().rows();
+  const auto nCols = matrix().cols();
+  const auto nNonZeros = matrix().nonZeros();
+  const auto baseWeights = makeMatrixMap(matrix());
+
+  // Note: need to store copy of weights as Eigen3 sorts compressed rows by j
+  // whereas eckit does not.
+  complexWeights_ = std::make_shared<ComplexMatrix>(nRows, nCols);
+  realWeights_ = std::make_shared<RealMatrix>(nRows, nCols);
+  auto complexTriplets = ComplexTriplets(nNonZeros);
+  auto realTriplets = RealTriplets(nNonZeros);
+
+  const auto sourceLonLats = array::make_view<double, 2>(source_.lonlat());
+  const auto targetLonLats = array::make_view<double, 2>(target_.lonlat());
+
+  geometry::UnitSphere unitSphere;
+
+  const auto setWeights = [&](auto i, auto j, const auto& baseWeight) {
+    const auto sourceLonLat =
+        PointLonLat(sourceLonLats(j, 0), sourceLonLats(j, 1));
+    const auto targetLonLat =
+        PointLonLat(targetLonLats(i, 0), targetLonLats(i, 1));
+
+    const auto alpha = unitSphere.greatCircleCourse(sourceLonLat, targetLonLat);
+
+    const auto deltaAlpha =
+        (alpha.first - alpha.second) * util::Constants::degreesToRadians();
+
+    const auto idx = std::distance(baseWeights.valuePtr(), &baseWeight);
+
+    complexTriplets[idx] = {int(i), int(j), std::polar(baseWeight, deltaAlpha)};
+    realTriplets[idx] = {int(i), int(j), baseWeight};
+  };
+
+  sparseMatrixForEach(setWeights, baseWeights);
+  complexWeights_->setFromTriplets(complexTriplets.begin(),
+                                   complexTriplets.end());
+  realWeights_->setFromTriplets(realTriplets.begin(), realTriplets.end());
+
+  ATLAS_ASSERT(complexWeights_->nonZeros() == matrix().nonZeros());
+  ATLAS_ASSERT(realWeights_->nonZeros() == matrix().nonZeros());
+}
+
+void SphericalVector::print(std::ostream&) const { ATLAS_NOTIMPLEMENTED; }
+
+void SphericalVector::do_execute(const FieldSet& sourceFieldSet,
+                                 FieldSet& targetFieldSet,
+                                 Metadata& metadata) const {
+  ATLAS_TRACE("atlas::interpolation::method::SphericalVector::do_execute()");
+
+  const auto nFields = sourceFieldSet.size();
+  ATLAS_ASSERT(nFields == targetFieldSet.size());
+
+  for (auto i = 0; i < sourceFieldSet.size(); ++i) {
+    do_execute(sourceFieldSet[i], targetFieldSet[i], metadata);
+  }
+}
+
+void SphericalVector::do_execute(const Field& sourceField, Field& targetField,
+                                 Metadata&) const {
+  ATLAS_TRACE("atlas::interpolation::method::SphericalVector::do_execute()");
+
+  const auto fieldType = sourceField.metadata().getString("type", "");
+  if (fieldType != "vector") {
+
+    auto metadata = Metadata();
+    Method::do_execute(sourceField, targetField, metadata);
+
+    return;
+  }
+
+  if (target_.size() == 0) {
+    return;
+  }
+
+  ATLAS_ASSERT_MSG(sourceField.variables() == 2 || sourceField.variables() == 3,
```

Comment:
Shouldn't this assertion come right at the beggining of the method?

Reply:
I'd also have to check that the field was a vector type. By this point in the function, it definitely is.